addons/mail/models/mail_activity.py

```diff
@@ -92,6 +92,7 @@ def _default_activity_type_for_model(self, model):
         'res.users', 'Assigned to',
         default=lambda self: self.env.user,
         index=True, required=True, ondelete='cascade')
+    tz = fields.Selection(string='Timezone', related="user_id.tz", store=True)
```

Comment:
Just to be that annoying guy, would rename to ``user_tz`` . Note that we will rewrite activities when user change their timezone, but I guess it does not happen frequently (and anyway you are not supposed to have millions of activities).